### DISCUSSION TITLE
libcurl Web Socket

### DISCUSSION THREAD

**Discussion by DerXanRam** [curl/curl#12016] [Q&A]
Hello guys i want to replicate this python code functionality using c++ and libcurl. The code subscribes blast api websocket service to get stream of pending transactions... here is the python code 

```
import asyncio
import json
from web3 import Web3
from websockets import connect

infura_ws_url = 'wss://bsc-mainnet.blastapi.io/xxxxxxxxxxxxxxxxxxxxxxxx'
infura_http_url = 'https://bsc-mainnet.blastapi.io/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
web3 = Web3(Web3.HTTPProvider(infura_http_url))

# Used if you want to monitor ETH transactions to a specific address
account = '0xcA143Ce32Fe78f1f7019d7d551a6402fC5350c73' 

async def get_event():
    async with connect(infura_ws_url) as ws:
        await ws.send('{"jsonrpc": "2.0", "id": 1, "method": "eth_subscribe", "params": ["newPendingTransactions"]}')
        subscription_response = await ws.recv()
        print(subscription_response)

        while True:
            try:
                message = await asyncio.wait_for(ws.recv(), timeout=15)
                response = json.loads(message)
                txHash = response['params']['result']
                print(txHash)
            except:
                pass

if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    while True:
        loop.run_until_complete(get_event())
```

if u ran this code u can clearly see stream of transaction hashes. I tried to to do this in c++ and building libcurl from source to enable websocket. I now it is expermental but i have no choice rather than to use it. So i tried by using [this example](https://curl.se/libcurl/c/websocket.html) and [this examples](https://curl.se/libcurl/c/websocket-cb.html) as reference but i can't.  I am new to c++ and curl Please help :pray:

**Comment by aimgo** [comment]
You need to provide details regarding how you built libcurl as well as output that demonstrates the unintended behavior otherwise there is little anyone can do to help you. This is better suited for stackoverflow.
